app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class VotesController < ApplicationController
+
+  def upvote 
+    @vote = Vote.find_by(id: params[:format])
```

Comment:
You're finding a vote here, not creating one with a work and a user.  This is a very unusual solution it works, but it means that for every work you're creating a vote with a list of users.